File: ios/XiriGoEcommerce/Feature/Home/Domain/Repository/HomeRepository.swift
import Foundation

// MARK: - HomeRepository

/// Repository protocol for fetching home screen data.
/// Implementations include `FakeHomeRepository` (sample data) and future `HomeRepositoryImpl` (Medusa API).
protocol HomeRepository: Sendable {
    func getBanners() async throws -> [HomeBanner]
    func getCategories() async throws -> [HomeCategory]
    func getPopularProducts() async throws -> [HomeProduct]
    func getDailyDeal() async throws -> DailyDeal?
    func getNewArrivals() async throws -> [HomeProduct]
    func getFlashSale() async throws -> FlashSale?
}

File: android/app/src/main/java/com/xirigo/ecommerce/feature/home/domain/repository/HomeRepository.kt
package com.xirigo.ecommerce.feature.home.domain.repository

import com.xirigo.ecommerce.feature.home.domain.model.DailyDeal
import com.xirigo.ecommerce.feature.home.domain.model.FlashSale
import com.xirigo.ecommerce.feature.home.domain.model.HomeBanner
import com.xirigo.ecommerce.feature.home.domain.model.HomeCategory
import com.xirigo.ecommerce.feature.home.domain.model.HomeProduct

interface HomeRepository {
    suspend fun getBanners(): List<HomeBanner>
    suspend fun getCategories(): List<HomeCategory>
    suspend fun getPopularProducts(): List<HomeProduct>
    suspend fun getDailyDeal(): DailyDeal?
    suspend fun getNewArrivals(): List<HomeProduct>
    suspend fun getFlashSale(): FlashSale?
}
